### PR TITLE
Compact mode finishing touches

### PR DESCRIPTION
**Pull request by jrainville**
This is based on top of https://github.com/status-im/status-desktop/pull/1759

It fixes all the points listed in that PR, expect the link hover. It shall be done in another PR

### CODE REVIEW

**Comment by simonricoo** [comment]
Hey I'm going to post some bug I have found it this PR:
Screen_Shot_2021-01-27_at_16_41_10

**Comment by simonricoo** [comment]
Screen_Shot_2021-01-27_at_16_41_10

**Design** 
Screen Shot 2021-01-28 at 10 25 27

**Comment by simonricoo** [comment]
Reactions container is miss-aligned. 

https://user-images.githubusercontent.com/52249767/106117770-a5fe0500-6153-11eb-95a2-21e3de3354e9.mov

**Comment by simonricoo** [comment]
Some glitch effect :D Not sure how to reproduce, I was just hovering around. 

https://user-images.githubusercontent.com/52249767/106117934-d6de3a00-6153-11eb-8831-4277653011c2.mov

**Comment by simonricoo** [comment]
Screen_Shot_2021-01-27_at_16_46_56

**Comment by simonricoo** [comment]
Screen_Shot_2021-01-27_at_16_44_56

**Comment by simonricoo** [comment]
Screen_Shot_2021-01-27_at_16_44_56

**Comment by simonricoo** [comment]
Screen_Shot_2021-01-27_at_16_46_47
Screen Shot 2021-01-28 at 11 12 28

Very bottom of this frame: https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/%F0%9F%92%AC-Chat%E2%8E%9CDesktop?node-id=4005%3A38331

**Comment by simonricoo** [comment]
Reaction tag top right corner looks wrong to me: 
Screen Shot 2021-01-28 at 11 18 00

**Comment by simonricoo** [comment]
The appearance has to be updated to, I would do a normal state here not this hover state to make it more clean, also it doesnt fit really well in the container now...
Screen_Shot_2021-01-28_at_12_06_41

**Comment by jrainville** [comment]
Thanks for the extensive list. Some of them I knew and are sadly limitations of the technology.

The mention bug is not related to this PR exactly, I spotted it on latest master as well. I'll check if there is not already an issue for that.

The missing padding on the mentions and missing rounded-corners is sadly something I can't do anything about. QML doesn't support the `display` css property nor `border-radius`, so I can't do that. If you find that it is too ugly as it is, we can go back to just changing the text color and maybe the font-weight?

The buttons container being cut at the top  and not aligned as in the design is something I tried long and hard to fix and I didn't find a good solution. I can try again and maybe with  a clear head I'll figure it out. Basically, the messages above have a higher Z index than the ones bellow for some weird reason and I don't think I have a way of saying to do the opposite. I tried reversing the order of the messages and it's still the same thing.

That's mostly it. The remaining bugs you found were either plain bugs or just stuff I hadn't noticed in the design. The above last two issues though are some very hard things to fix with what QML offers so don't be surprised if it can't be 100% exactly as you envisioned

File: ui/app/AppLayouts/Chat/ChatColumn/Message.qml [hunk 1]
```diff
@@ -36,7 +37,8 @@ Item {
     property bool isStatusMessage: contentType === Constants.systemMessagePrivateGroupType
     property bool isSticker: contentType === Constants.stickerType
     property bool isText: contentType === Constants.messageType
-    property bool isMessage: isEmoji || isImage || isSticker || isText || isAudio || contentType === Constants.communityInviteType
+    property bool isMessage: isEmoji || isImage || isSticker || isText || isAudio
```
Comment (richard-ramos) [inline ui/app/AppLayouts/Chat/ChatColumn/Message.qml:40]: This list keeps growing!

**Comment by johnlea-quiup** [comment]
@jrainville re.  rounded corners and even enterally custom shapes, we did this in many places when I was working on Ubuntu Unity8 which used QML for everything, most notably on our icons which were a totally custom shape.   In case it helps with implementing rounded corners, here are some links about it was done

https://phone.docs.ubuntu.com/en/apps/api-qml-current/QtQuick.qtquick-scenegraph-openglunderqml-example

https://github.com/Sturmflut/ubuntu-touch-squircle-example

https://api-docs.ubports.com/sdk/apps/qml/QtQuick/Qt%20Quick%20Examples%20-%20Canvas.html

File: src/status/libstatus/chat.nim [hunk 2]
```diff
@@ -63,11 +63,16 @@ proc chatMessages*(chatId: string, cursor: string = ""): (string, seq[Message])
...
+  
+  let pk = status_settings.getSetting[string](Setting.PublicKey, "0x0")
 
   let rpcResult = parseJson(callPrivateRPC("chatMessages".prefix, %* [chatId, cursorVal, 20]))["result"]
+  var msg: Message
   if rpcResult["messages"].kind != JNull:
     for jsonMsg in rpcResult["messages"]:
-      messages.add(jsonMsg.toMessage)
+      msg = jsonMsg.toMessage
+      msg.hasMention = concat(msg.parsedText.map(t => t.children.filter(c => c.textType == "mention" and c.literal == pk))).len > 0
```
Comment (richard-ramos) [inline src/status/libstatus/chat.nim:74]: Could this code be part of `toMessage`, and have toMessage receive the public key?

Comment (jrainville) [reply src/status/libstatus/chat.nim:74]: Yeah I can try that

**Comment by jrainville** [comment]
Alright, I fixed all the issues you found @simonricoo that I could fix easily and are not limited by the technology. Let me know if you find something else or if the remaining issues are too ugly to leave as is. We can always find another way to present things.

File: src/status/libstatus/chat.nim (hunk 2 above)
Comment (jrainville) [reply src/status/libstatus/chat.nim]: @richard-ramos Updated as you suggested. It worked perfectly!

File: src/status/signals/messages.nim [hunk 3]
```diff
@@ -277,6 +278,8 @@ proc toMessage*(jsonMsg: JsonNode): Message =
       signature: jsonMsg["commandParameters"]["signature"].getStr
     )
 
+  message.hasMention = concat(message.parsedText.map(t => t.children.filter(c => c.textType == "mention" and c.literal == pk))).len > 0
```
Comment (richard-ramos) [inline src/status/signals/messages.nim:281]: Smart! :thinking:

**Review by richard-ramos — APPROVED** [review]

**Pull request merged**